app/assets/src/components/views/SampleUploadFlow/UploadSampleStep.jsx

```diff
@@ -531,35 +532,42 @@ class UploadSampleStep extends React.Component {
       ...(sampleType === "local" ? { removedLocalFiles: [] } : {}),
     });
 
-    const samplesWithSelectIds = this.addSelectIdToSamples(samples);
+    const newSamplesWithSelectIds = this.addSelectIdToSamples(newSamples);
 
     const {
-      samples: validatedSamples,
+      samples: validatedNewSamples,
       removedLocalFiles,
-    } = await this.validateAddedSamples(samplesWithSelectIds, sampleType);
+    } = await this.validateAddedSamples(newSamplesWithSelectIds, sampleType);
 
-    const newSamples = this.mergeSamples(
+    // This de-dupes the newly added samples using the sample name. The older sample's selectId is used.
+    const mergedSamples = this.mergeSamples(
       this.state[samplesKey],
-      validatedSamples
+      validatedNewSamples
     );
 
-    const newSelectedSampleIds = new Set(
-      union(
-        Array.from(this.state[selectedSampleIdsKey]),
-        map(SELECT_ID_KEY, validatedSamples)
-      )
+    const mergedSampleIds = map(SELECT_ID_KEY, mergedSamples);
+
+    // Get the selectIds of all newly added samples that weren't already added.
+    const dedupedNewSampleIds = intersection(
```

Comment:
Why call this deduped? according to the description its just `newlyAddedSampleIds`.

Reply:
Makes sense! Will fix.